deployment/dataproc/gtfs-feed-fetcher/migrate_feeds_to_otpv1.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python
+"""Modifies GTFS files to OTP v1 standard.
+
+OTPv2 updates as GTFS standard updates, however not backported to OTPv1.
+We can't migrate to OTPv2, so manually modify GTFS for OTPv1 support level.
+"""
+import logging
+import zipfile
+import os
+
+logging.basicConfig()
+LOG = logging.getLogger()
+LOG.setLevel(logging.INFO)
+
+gtfs_dir = 'gtfs'
+
+def unzip_file(filename):
+    feed_path = gtfs_dir + "/" + filename + ".zip"
+    if zipfile.is_zipfile(feed_path):
+        with zipfile.ZipFile(feed_path, 'r') as zip_ref:
+            zip_ref.extractall(gtfs_dir)
+            LOG.debug('Unzipped %s', filename)
+            return
+    else:
+        LOG.warn('File %s does not look like a valid zip file.', filename)
+
+def clean_up_txt_files():
+    for root, _, files in os.walk(gtfs_dir):
+        for file in files:
+            if file.endswith('.txt'):
+                os.remove(os.path.join(root, file))
+    LOG.debug('Cleaned up .txt files')
+
+def re_zip_files(filename):
+    feed_path = gtfs_dir + "/" + filename + ".zip"
+    with zipfile.ZipFile(feed_path, 'w', zipfile.ZIP_DEFLATED) as zip_ref:
+        for root, _, files in os.walk(gtfs_dir):
+            for file in files:
+                if file.endswith('.txt'):
+                    file_path = os.path.join(root, file)
+                    zip_ref.write(file_path, os.path.relpath(file_path, gtfs_dir))
+        LOG.debug('Re-zipped %s', feed_path)
+
+def modify_septa_bus():
+    # New GTFS includes replacement trolleybuses, unsupported transit type in OTPv1
+    # Convert to buses to prevent error building OTP
+    # Reference: https://github.com/septadev/GTFS?tab=readme-ov-file#routestxt
+    txt_path_to_update = os.path.join(gtfs_dir, 'routes.txt')
+    content=''
+    try:
+        with open(txt_path_to_update, 'r') as file:
+            content = file.read()
+            content = content.replace(',,11,,', ',,3,,')
+        with open(txt_path_to_update, 'w') as file:
+            file.write(content)
+        LOG.debug('Modified %s', txt_path_to_update)
+    except Exception as e:
+        raise ValueError('Error modifying %s: %s', txt_path_to_update, e)
```

Comment:
This is a little fragile. These .txt files are actually formatted as CSV, so we should be able to use a CSV reader and update the `route_type` column. However, this is working well enough for now, and we do have an `except` that will pause on this failure, so perhaps this can be made more robust in the future, when it fails.

<img width="1221" alt="image" src="https://github.com/user-attachments/assets/4045eecd-b609-4a83-8ba0-3c294ca77d34" />

Reply:
This is a great point, thank you! This is definitely a file subject to change with the new SEPTA rollout so it would benefit to be robust. Documenting that we talked briefly about solutions, I ended up moving forward with a csv reader and iterating through the rows and replacing values in the correct column as needed. This is a smaller file (~10kb, <200 rows) so bringing in a library to replace these values more efficiently doesn't feel worth it. Here is t[hat new step in the latest successful build ](https://github.com/azavea/cac-tripplanner/actions/runs/14480346972/job/40615799660)(note I pushed up a temp commit to test this job but that prevents building the graph). Let me know what you think!